.github/scripts/delete_old_branches.py

```diff
@@ -0,0 +1,183 @@
+            PR_WINDOW
+            and now - convert_gh_timestamp(pr_infos[-1]["updatedAt"]) > PR_WINDOW
+        ):
+            break
+
+    # Get the most recent PR for each branch base (group gh together)
+    prs_by_branch_base = {}
+    for pr in pr_infos:
+        pr["updatedAt"] = convert_gh_timestamp(pr["updatedAt"])
+        branch_base_name = pr["headRefName"]
+        if x := re.match(r"(gh\/.+)\/(head|base|orig)", branch_base_name):
+            branch_base_name = x.group(1)
+        if branch_base_name not in prs_by_branch_base:
+            prs_by_branch_base[branch_base_name] = pr
+        else:
+            if pr["updatedAt"] > prs_by_branch_base[branch_base_name]["updatedAt"]:
+                prs_by_branch_base[branch_base_name] = pr
+    return prs_by_branch_base
+
+
+def delete_branch(repo: GitRepo, branch: str) -> None:
+    repo._run_git("push", "origin", "-d", branch)
+
+
+def delete_branches() -> None:
+    now = datetime.now().timestamp()
+    git_repo = GitRepo(str(REPO_ROOT), "origin", debug=True)
+    branches = get_branches(git_repo)
+    prs_by_branch = get_prs()
+
+    delete = []
+    # Do not delete if:
+    # * associated PR is open, closed but updated recently, or contains the magic string
+    # * no associated PR and branch was updated in last 1.5 years
+    # * is protected
+    # Setinng different values of PR_WINDOW will change how branches with open
+    # PRs are treated depending on how old the branch is.  The default value of
+    # 90 will allow branches with open PRs to be deleted if the PR hasn't been
+    # updated in 90 days and the branch hasn't been updated in 1.5 years
+    for base_branch, (date, sub_branches) in branches.items():
+        print(f"[{base_branch}] Updated {(now - date) / SEC_IN_DAY} days ago")
+        pr = prs_by_branch.get(base_branch)
+        if pr:
+            print(
+                f"[{base_branch}] Has PR {pr['number']}: {pr['state']}, updated {(now - pr['updatedAt']) / SEC_IN_DAY} days ago"
+            )
+            if PR_BODY_MAGIC_STRING in pr["body"]:
+                continue
+            if pr["state"] == "OPEN":
+                continue
+            if (
+                now - pr["updatedAt"] < CLOSED_PR_RETENTION
+                or (now - date) < CLOSED_PR_RETENTION
+            ):
+                continue
+        elif now - date < NO_PR_RETENTION:
+            continue
+        print(f"[{base_branch}] Checking for branch protections")
+        if any(is_protected(sub_branch) for sub_branch in sub_branches):
+            print(f"[{base_branch}] Is protected")
+            continue
+        for sub_branch in sub_branches:
+            print(f"[{base_branch}] Deleting {sub_branch}")
+            delete.append(sub_branch)
+        if ESTIMATED_TOKENS[0] > 400:
+            print("Estimated tokens exceeded, exiting")
+            break
+
+    print(f"To delete ({len(delete)}):")
+    for branch in delete:
+        print(branch)
+    for branch in delete:
```

Comment:
Nit (will help with debugging if script will crash at some point)
```suggestion
        print(f"About to delete {branch}")
```